src/db/db.c

```diff
@@ -72,11 +72,14 @@ dbNew(PgClient *client, ProtocolClient *remoteClient, const String *applicationN
     MEM_CONTEXT_NEW_BEGIN("Db")
     {
         this = memNew(sizeof(Db));
-        this->memContext = memContextCurrent();
 
-        this->client = pgClientMove(client, this->memContext);
-        this->remoteClient = remoteClient;
-        this->applicationName = strDup(applicationName);
+        *this = (Db)
+        {
+            .memContext = memContextCurrent(),
+            .client = pgClientMove(client, memContextCurrent()),
+            .remoteClient = remoteClient,
```

Comment:
In dbNew, it might be clearer to set .memContext = MEM_CONTEXT_NEW() and then pull out .client below the initializer and have it as `this->client = pgClientMove(client, this->memContext)`